From f456420bd5158edd015c1303b29aaf67253f44fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20R=C3=BCberg?=
 <22551563+Toorero@users.noreply.github.com>
Date: Thu, 19 Aug 2021 23:19:41 +0200
Subject: [PATCH 1/4] Changed to proper generic python command

---

Obtained from: https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/345
Minor patch changes by Orson Teodoro <orsonteodoro@hotmail.com>
Document changes for Apache-2.0 compliance.

License header for
BlenderPkg/build_linux.sh:
build.py:
src/bindings/pyrpr/rpr.py:

#**********************************************************************
# Copyright 2020 Advanced Micro Devices, Inc
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#********************************************************************

Toorero is not a mispelling of Teodoro.  We are two different people.
Teodoro edited the patch for newer point releases and to complete the patch.

--- a/BlenderPkg/build_linux.sh.orig	2025-08-21 12:13:29.000000000 -0700
+++ b/BlenderPkg/build_linux.sh	2025-09-30 14:55:34.317881228 -0700
@@ -15,8 +15,10 @@
 # limitations under the License.
 #********************************************************************
 
+# oteodoro: Changed for generic python3 call for systems without python3.7.
+
 pushd ..
 ./build.sh
 popd
 
-python3.11 create_zip_addon.py
+python3 create_zip_addon.py		# oteodoro: python3.11 -> python3
--- a/build.py.orig	2025-08-21 12:13:29.000000000 -0700
+++ b/build.py	2025-09-30 14:59:36.794003387 -0700
@@ -40,7 +40,7 @@ subprocess.check_call([sys.executable, '
 subprocess.check_call([sys.executable, 'rpr_load_store.py'])
 os.chdir(cwd)
 
-if sys.version_info.major == 3 and sys.version_info.minor == 11:
+if True or sys.version_info.major == 3 and sys.version_info.minor == 11:					# oteodoro - changed line
     # we are going to build RPRBlenderHelper only for python 3.11
     os.chdir('RPRBlenderHelper')
     shutil.rmtree('.build', ignore_errors=True)
--- a/src/bindings/pyrpr/rpr.py.orig	2025-09-30 14:53:24.639955827 -0700
+++ b/src/bindings/pyrpr/rpr.py	2025-09-30 15:00:40.972977273 -0700
@@ -97,7 +97,7 @@ def export(json_file_name, dependencies,
         if distro.id() != 'gentoo':									# oteodoro added line
           cffi_libs_dir = Path(_cffi_backend.__file__).parent / '.libs_cffi_backend'			# oteodoro changed indent
           if not cffi_libs_dir.is_dir():								# oteodoro changed indent
-              cffi_libs_dir = Path(_cffi_backend.__file__).parent / 'cffi.libs'			# oteodoro changed indent
+              cffi_libs_dir = Path(_cffi_backend.__file__).parent / 'cffi'				# oteodoro changed indent, Toorero: Added alternative cffi_libs_dir location 46be6f1
 
           if cffi_libs_dir.is_dir():									# oteodoro changed indent
               for path in cffi_libs_dir.iterdir():							# oteodoro changed indent
